web/packages/teleterm/src/ui/ModalsHost/modals/AuthenticateWebDevice/AuthenticateWebDevice.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Alert from 'design/Alert';
+import { ButtonPrimary, ButtonSecondary } from 'design/Button';
+import Dialog, { DialogContent } from 'design/Dialog';
+import Flex from 'design/Flex';
+import { useAsync } from 'shared/hooks/useAsync';
+
+type Props = {
+  onClose(): void;
+  onAuthorize(): void;
+};
+
+export const AuthenticateWebDevice = ({ onAuthorize, onClose }: Props) => {
+  const [attempt, run] = useAsync(async () => {
+    await onAuthorize();
+    onClose();
+  });
+
+  return (
+    <Dialog open={true}>
+      <DialogContent>
+        Would you like to launch an authorized web session?
```

Comment:
I didn't test how it actually looks, but I think it's missing some details as to which cluster this is about, no?

I know that we're already switching workspaces, but I think in modals like that it's worth repeating the name of the cluster. Modals are shown with this dark overlay, so it's better if the name is included somewhere in the modal. Look up the stories for other modals for examples.

Reply:
Sure I'll add it in a header or something. This was just what was designed in the figma. I'll push back with some feedback to design